code/modules/mob/living/living.dm

```diff
@@ -124,6 +124,12 @@
 					if(!(world.time % 5))
 						to_chat(src, span_warning("[L] is restraining [P], you cannot push past."))
 					return TRUE
+		if(length(gunpointed))
+			to_chat(src, span_warning("[L] is being held at gunpoint, it's not wise to push [L.p_them()]."))
+			return TRUE
+		if(L.GetComponent(/datum/component/gunpoint))
+			to_chat(src, span_warning("[L] is holding someone at gunpoint, you cannot push past."))
+			return TRUE
```

Comment:
These should be handled in the examine signal on the component. This is absolutely not acceptable

Reply:
This is not for examine, but for Bump logic. Right now the system is cheesable by the third person just bumping onto the target, triggering the shot. How can I make the target not pushable?

Reply:
My bad, I didn't see which proc it was. You should allow `COMSIG_LIVING_MOB_BUMP` to return a flag saying to blog the bump.

Reply:
Uh, how exactly can I return the flag and block the bump?